signer_test.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+
+	p2pcrypto "github.com/libp2p/go-libp2p-crypto"
+)
+
+const signerData = `
+{"index":{"_index":"mesh_outside","_type":"docker"}}
+{"@timestamp":"2019-06-30T11:07:15.000Z", "source":"stderr", "log":"{\"error\":\"Post http://ganache:8545: EOF\",\"level\":\"error\",\"msg\":\"BlockWatcher error encountered\",\"time\":\"2019-06-30T11:07:15Z\"}", "container_id":"b33055fd8054ad060d167da59798621a126093582e31c3c273ad2ddff928cc1b", "container_name":"/docker-compose-shipper_mesh_1"}
+`
+
+func TestSerializeAndDeserialize(t *testing.T) {
+	pubKeyHex := "036a775c4db73fd351191d0a0e19862ecbb70cbe2626097adfb70ffd4f9ea081bf"
+	pubBytes, err := hex.DecodeString(pubKeyHex)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pub, err := p2pcrypto.UnmarshalSecp256k1PublicKey(pubBytes)
+	rawPubKey, err := pub.Raw()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	returnedPub := hex.EncodeToString(rawPubKey)
+	if returnedPub != pubKeyHex {
+		t.Fatal("not matching public keys")
+	}
+}
+
+func TestLoadPrivateKey(t *testing.T) {
+	signer, err := NewP2PSignerFromKeyPath("./__fixtures__/privkey")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Println(signer.PubKey())
```

Comment:
Let's actually assert something here.

Reply:
added a simple assertion to an expected public key.